.github/workflows/build_x86.yml

```diff
@@ -667,6 +667,17 @@ jobs:
           -DOSQUERY_NO_DEBUG_SYMBOLS=${{ steps.debug_symbols_settings.outputs.VALUE }} \
           ${{ steps.build_paths.outputs.SOURCE }}
 
+    - name: Insert version string in Info.plist
+      shell: bash
+
+      env:
+        CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+
+      run: |
+        osquery_version=$(cd ${{ steps.build_paths.outputs.SOURCE }} && git describe --tags --always )
+
+        plutil -insert CFBundleShortVersionString -string $osquery_version workspace/src/tools/deployment/macos_packaging/Info.plist
```

Comment:
Shouldn't need `workspace/src`, instead it's better to set the directory. It might already be set?

Reply:
I tried `tools/deployment/macos_packaging/Info.plist` but that didn't fly. Not entirely sure what you mean about setting the directory, I'm more familiar with GitLab's CI syntax than GitHub's. Is there a variable or something I can use so that path isn't hardcoded?

Reply:
Digging into this, back on https://github.com/osquery/osquery/blob/153f9591fc43a6ea4f6f56e26a410e92c37e1599/.github/workflows/build_x86.yml#L514 sets it as SOURCE. Which is then accessed as `${{ steps.build_paths.outputs.SOURCE }}`. The other steps here have some examples.

But, saying all that, I think it's weird to modify the source. I'm not sure what the right idiom is. It might be better suited for the cmake `install_directives.cmake`

Reply:
Trying to do it through cmake seems like a six vs. half dozen situation to me. I could pass the version into cmake in the "Configure the Project" step (L655), but then I'd need to figure out how to use cmake to insert a key/value into a plist.

Reply:
@macjustice We would really like this to be reproducible also outside of the CI or additional manual steps.

There are a couple of ways I can think one can do this:
1. If the modification of the plist is easy to be done manually to, `tools/deployment/macos_packaging/Info.plist` can become a template for CMake to use and generate the plist from via `configure_file`
2. Run that plutil command through CMake via an `install(CODE [...])` directive, which uses `execute_process`.

Reply:
OK, I follow. I'll take a swing at that, thanks for the clarification!